pkgs/build-support/trivial-builders.nix

```diff
@@ -318,23 +321,46 @@ rec {
   symlinkJoin =
     args_@{ name
          , paths
+         , outputs ? [ "out" ]
          , preferLocalBuild ? true
          , allowSubstitutes ? false
          , postBuild ? ""
          , ...
          }:
     let
+      secondaryOutputs = builtins.filter (o: o != "out") outputs;
+      outputToPathsAttr = output: {
+        name = output + "Paths";
+        value = lib.flatten (map (p: p.${output} or []) paths);
+      };
       args = removeAttrs args_ [ "name" "postBuild" ]
         // {
           inherit preferLocalBuild allowSubstitutes;
-          passAsFile = [ "paths" ];
-        }; # pass the defaults
+          outputs = [ "out" ] ++ secondaryOutputs;
+          passAsFile = [ "paths" ] ++ map (o: o + "Paths") secondaryOutputs;
+        } // builtins.listToAttrs (map outputToPathsAttr secondaryOutputs);
     in runCommand name args
       ''
         mkdir -p $out
         for i in $(cat $pathsPath); do
           ${lndir}/bin/lndir -silent $i $out
         done
+
+        for output in ${toString secondaryOutputs}; do
+          eval output_path=\$$output
```

Comment:
Rather then eval, you can use indirect variable access: `output_path=${!output}`: https://tldp.org/LDP/abs/html/bashver2.html#EX78

Reply:
done!